### PR TITLE
Ensure generated SDK changes reach release notes

### PR DESCRIPTION
**Pull request by sjmiller609**
## Summary

Automatically makes every generated SDK change releasable. Explicit `feat:`, `fix:`, `perf:`, and `revert:` commit types are preserved; all other SDK-affecting merges are classified as features instead of being silently omitted by release-please.

## Validation

- actionlint passed
- shell normalization checks cover untyped, `api:`, `fix:`, and breaking `feat!:` titles
- the PR workflow runs real generation and SDK validation

---

> [!NOTE]
> **Low Risk**
> Changes only CI commit-message shaping for SDK staging pushes; wrong classification could affect semver/changelog but not application runtime.
> 
> **Overview**
> SDK pushes on **main** now run merged PR titles (or the commit subject) through **`normalize-sdk-commit-message.sh`** before `stlc build --commit`, so **release-please** always sees a conventional type instead of skipping untyped titles.
> 
> The normalizer **keeps** explicit `feat` / `fix` / `perf` / `revert` messages (including scoped and `!` breaking forms). **Non-matching** titles get a **`feat:`** prefix; titles that look like a custom breaking type (`type!:`) are rewritten to **`feat!:`** plus the original text.
> 
> **`stlc-generate.yml`** runs the shell test script in the generate job and watches the script paths on PR/push. The SDK generation runbook documents that every merged API/SDK PR is treated as releasable with this classification.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 295831b6568e6da110d08e36f37702224da75a5e. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: .github/workflows/stlc-generate.yml [hunk 1]
```diff
@@ -199,6 +200,10 @@ jobs:
           set -euo pipefail
           msg=$(gh api "repos/$REPO/commits/$SHA/pulls" --jq '.[0].title // empty')
           [ -n "$msg" ] || msg=$(git log -1 --pretty=%s)
+          conventional='^(feat|fix|perf|revert)(\([^)]*\))?!?:[[:space:]]+'
+          if [[ ! "$msg" =~ $conventional ]]; then
+            msg="feat: $msg"
+          fi
```
[the remark this answers is not in the document]
Comment (sjmiller609) [reply .github/workflows/stlc-generate.yml]: Fixed. Unknown `!:` types now normalize to `feat!:` so release-please preserves the breaking-change bump; recognized conventional breaking types remain unchanged.

**Review by masnwilliams — DISMISSED** [review]
requesting changes for one release-policy correctness issue:

`.github/workflows/stlc-generate.yml:203-210` reimplements release-please's Conventional Commit grammar, but the grammars disagree. `api/v2!:` and `api.sdk!:` are valid breaking headers to release-please, yet the `breaking` regex misses them and rewrites them as non-breaking features. Conversely, `feat():` matches `conventional` even though release-please rejects the empty scope, so that commit can still be omitted entirely.

Please make these patterns match the actual parser boundary—nonempty scopes without nested parentheses and types excluding only whitespace/`()`/`!`/`:`—and check in table-driven tests for these boundary cases. This release policy is too consequential to leave as manually tested inline shell.

For example:

```bash
conventional='^(feat|fix|perf|revert)(\([^()]+\))?!?:[[:space:]]+'
breaking='^[^()!:[:space:]]+(\([^()]+\))?!:[[:space:]]+'
```

**Review by rgarcia — APPROVED** [review]

**Comment by sjmiller609** [comment]
Addressed Mason’s parser-contract feedback in f07d211. The normalizer now accepts zero whitespace, rejects empty/nested scopes, supports the parser’s type boundary, and preserves breaking markers when remapping types. Committed table-driven tests cover the requested boundary cases and run in the generation job.

File: .github/scripts/normalize-sdk-commit-message.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+
+normalize_sdk_commit_message() {
+  local msg=$1
+  local conventional='^(feat|fix|perf|revert)(\([^()]+\))?!?:[[:space:]]*'
+  local breaking='^[^()!:[:space:]]+(\([^()]+\))?!:[[:space:]]*'
+
+  if [[ "$msg" =~ $conventional ]]; then
+    printf '%s\n' "$msg"
```
[the remark this answers is not in the document]
Comment (sjmiller609) [reply .github/scripts/normalize-sdk-commit-message.sh:9]: This is intentional and matches the pinned parser. release-please v17.3.0 uses `@conventional-commits/parser` v0.4.1, whose summary grammar is `":" <whitespace>* <text>` and whose implementation calls `whitespace(scanner)` optionally. It therefore accepts zero whitespace after `:`. The no-space cases preserve valid release-please input and are covered by the committed tests.
